### ISSUE TITLE
Where is the training parameter `epsilon`?

### ISSUE THREAD

**Issue by aatifjiwani** [imperial-qore/TranAD#63]
In section 3.4, equation 10 of the paper, the authors introduce the overall loss function that uses a training parameter `epsilon` that is scaled to the power of `-n` where `n` is the training epoch. However, I do not see the presence of `epsilon` anywhere in the code - it seems `e^(-n)` was replaced with `1 / n` in the code

https://github.com/imperial-qore/TranAD/blob/566f8c8ba58ad605ab157671b1a9ca7ee485a7a8/main.py#L264

**Comment by patrickfleith** [comment]
Yes, I think this is because it is an evolution from the code of the original paper. However, the behavior is the same. However, it would be great to have comment from the authors on why they made this decision to change it would be informative :)

Let. `epsilon` be a float close but greater than 1, for example 1.2:
```python
epsilon = 1.2
[epsilon**(-n) for n in range(1,11) ]
```
We get 
```
[0.8333333333333334,
 0.6944444444444445,
 0.5787037037037037,
 0.4822530864197532,
 0.40187757201646096,
 0.3348979766803842,
 0.2790816472336535,
 0.2325680393613779,
 0.19380669946781492,
 0.1615055828898458]
```

With what is implemented in the code we have
```python 
[(1/n) for n in range(1,11)]
```

We get 
```
[1.0,
 0.5,
 0.3333333333333333,
 0.25,
 0.2,
 0.16666666666666666,
 0.14285714285714285,
 0.125,
 0.1111111111111111,
 0.1]
```

**Comment by patrickfleith** [comment]
I checked #45 there is another branch of the repo where the loss function is different.

```python
z = model(window, elem)
l1 = l(z, elem) if not isinstance(z, tuple) else (0.9 ** n) * l(z[0], elem) + (1 - 0.9 ** n) * l(z[1], elem)
l1 += 0 if not isinstance(z, tuple) or len(z) < 3 else  (0.9 ** n) * l(z[2], elem) - (1 - 0.9 ** n) * l(z[1], elem)
```
Again as a similar deacrising behaviour, but still not not e^(-n) but e^(n). Choosing the sign of the exponent (n or -n) seems to be a matter of wether you choose e slighly greater or slightly lower than 1.

**Comment by patrickfleith** [comment]
Not sure which loss function is valid for the best performances. @shreshthtuli Could you tell us if it is advice to use the loss from the current `main` branch or from. the `loss` branch?

**Comment by shreshthtuli** [comment]
The version used for the paper had e^(-n) decay; however, we observed decay of 1/n made training slightly more robust and hence use this in the latest code.
